### PR TITLE
Fix features since nightly-2018-08-14 disallowed old features

### LINKED ISSUE #20: Add #![feature(alloc)], when enabling collections and/or alloc feature
error[E0658]: use of unstable library feature 'alloc': this library is unlikely to be stabilized in its current form or name (see issue #27783)
  --> /home/mathias/.cargo/registry/src/github.com-1ecc6299db9ec823/core_io-0.1.20190427/src/lib.rs:15:51
   |
15 | #[cfg(all(feature="collections",no_collections))] extern crate alloc as collections;
   |                                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(alloc)] to the crate attributes to enable

### PR DESCRIPTION
**Pull request by parasyte**
Fixes #20

### CODE REVIEW

**Comment by parasyte** [comment]
@jethrogb It worked‼️

**Comment by parasyte** [comment]
Some additional notes: I added `feature(nll)` with `bind_by_move_pattern_guards` only because it's required starting with nightly-2019-02-25. I think https://github.com/rust-lang/rust/issues/59192 removed the need for the `nll` feature with `bind_by_move_pattern_guards`, but I haven't tested. And since `nll` is still accepted by recently nightlies, I didn't see a reason to add more conditions.

This *might* be further addressed by https://github.com/rust-lang/rust/issues/57895 in the future ...

**Comment by parasyte** [comment]
Ping @jethrogb I would like to get this merged so I can also address #19; I have a patch staged for that, it depends on fixing the build (this PR).

**Comment by parasyte** [comment]
@jethrogb Sorry to bother, but this is ready for review.

I'm also considering taking up the maintainer role (#15) since I'm planning on depending on this crate soon. This might be a bit off-topic for the ticket, but I will need something like `core_io` until I can port `std` to my target. And I don't expect that to happen any time soon. The portability WG repo hasn't seen a lot of activity lately. :)

File: build.rs [hunk 1]
```diff
@@ -52,6 +52,11 @@ type Date = &'static str;
 /// versions for which to enable particular features.
 type ConditionalCfg = (Cfg, &'static [(Date, Cfg)]);
 const CONDITIONAL_CFGS: &'static [ConditionalCfg] = &[
+	(None, &[("2019-02-24", Some("pattern_guards"))]),
+	(None, &[("2018-08-14", Some("non_exhaustive"))]),
+	(Some("collections"), &[("2018-08-13", None)]),
```
Comment (jethrogb) [inline build.rs]: Isn't this a dupe of `no_collections`?

File: build.rs [hunk 2]
```diff
@@ -52,6 +52,11 @@ type Date = &'static str;
...
+	(Some("str_char"), &[("2018-08-13", None)]),
+	(Some("unicode"), &[("2018-08-13", None)]),
```
Comment (jethrogb) [inline build.rs:57]: Combine this into a single cfg.

**Review by jethrogb — COMMENTED** [review]
Sorry for the delay. Mostly LGTM, but a few comments.

File: build.rs (hunk 1 above)
Comment (parasyte) [reply build.rs]: Good point! Yeah, I think `not(no_collections)` will do it. Testing this...

Comment (parasyte) [reply build.rs]: Well, I cannot use the oldest versions of rust-nightly on macOS; I let `nightly-2016-03-11` run all night and it never finished. I'm running the test again starting with `nightly-2016-12-05`.

Comment (parasyte) [reply build.rs]: AH! Yep, it works.

File: build.rs (hunk 2 above)
Comment (parasyte) [reply build.rs:57]: `str_char` and `unicode` features are now enabled by the `unicode` cfg.

**Pull request merged**